src/main/java/controllers/ConsoleInputController.java

```diff
@@ -99,48 +102,66 @@ private void checkRecurring() {
      * @param input : Input typed by user into CLI
      */
     @Override
-    public void onCommandReceived(String input) {
+    public void onCommandReceived(String input) throws DukeException {
```

Comment:
which line of code throws DukeException? Refactor it such that it doesn't throw DukeException because it doesn't make sense for View to handle exceptions

Reply:
why not? Considering ConsoleInputController is kind or like a parser, it could run into errors exceptions when reading input, for which something else can handle the exception (CLIView perhaps?). By throwing the exception something else can handle it later no? please advise i am unsure

Reply:
if ConsoleInputController is a parser, then it should handle its own exceptions and not throw to others to handle. that's why our controller has try catch statements inside to catch the relevant exceptions